src/script/index.js

```diff
@@ -24,10 +26,58 @@ for (var i = 0; i < paramPairsLength; i++) {
 delete i
 delete paramPair
 
+function add_instance(text, disabled, selected) {
+    const opt = document.createElement('option');
+    opt.innerText = text;
+    if (disabled) {
+        opt.setAttribute('disabled', true);
+    }
+    if (selected) {
+        opt.setAttribute('selected', true);
+    }
+    choose_instance.appendChild(opt);
+}
+
 if (prefillInstance != null) {
-    document.getElementById('instance').value = normalizeUrl(prefillInstance);
+    const url = normalizeUrl(prefillInstance);
+    instance.value = url;
+    add_instance(url.slice(8, url.length - 1), false, true);
```

Comment:
How about this:

```suggestion
    add_instance(url.split('/')[2], false, true);
```

Since domains with `http://` also count as normalized, we can't be sure that the prefix will be 8 characters and not 7

Reply:
I went for properly parsing the URL using [jlong's trick](https://gist.github.com/jlong/2428561) - let me know what you think.

Reply:
Resolved by 06f7c372b5bd2c3650afbe8a0b61289ba5a9e22d